qp_shogun/sortmerna/sortmerna.py

```diff
@@ -0,0 +1,180 @@
+#    '{0}index/silva-euk-18s-id95.idx:'
+#    '{0}rRNA_databases/silva-euk-28s-id98.fasta,'
+#    '{0}index/silva-euk-28s-id98.idx:'
+#    '{0}rRNA_databases/rfam-5s-database-id98.fasta,'
+#    '{0}index/rfam-5s-database-id98.idx:'
+#    '{0}rRNA_databases/rfam-5.8s-database-id98.fasta,'
+#    '{0}index/rfam-5.8s-database-id98.idx'
+# ).format(DIR)
+
+SORTMERNA_PARAMS = {
+    'a': 'Number of threads'}
+
+
+def generate_sortmerna_commands(forward_seqs, reverse_seqs, map_file,
+                                out_dir, parameters):
+    """Generates the Sortmerna commands
+    Parameters
+    ----------
+    forward_seqs : list of str
+        The list of forward seqs filepaths
+    reverse_seqs : list of str
+        The list of reverse seqs filepaths
+    map_file : str
+        The path to the mapping file
+    out_dir : str
+        The job output directory
+    parameters : dict
+        The command's parameters, keyed by parameter name
+    Returns
+    -------
+    cmds: list of str
+        The Sortmerna commands
+    samples: list of tup
+        list of 4-tuples with run prefix, sample name, fwd read fp, rev read fp
+    Notes
+    -----
+    Currently this is requiring matched pairs in the make_read_pairs_per_sample
+    step but implicitly allowing empty reverse reads in the actual command
+    generation. This behavior may allow support of situations with empty
+    reverse reads in some samples, for example after trimming and QC.
+    """
+    # matching filenames, samples, and run prefixes
+    samples = make_read_pairs_per_sample(forward_seqs, reverse_seqs, map_file)
+
+    cmds = []
+    # param_string = _format_params(parameters, SORTMERNA_PARAMS)
+
+    # Sortmerna 2.1 does not support processing of
+    # compressed files but they said the newest release might
+    # but that version first has to be tested before use and currently
+    # does not have MAC OS supported release
+
+    threads = parameters['Number of threads']
+
+    for run_prefix, sample, f_fp, r_fp in samples:
+        if r_fp is None:
+            cmds.append('sortmerna --ref %s --reads %s '
+                        '--aligned %s --other %s '
+                        '--fastx --blast 1 --num_alignments 1 -a %s' % (
+                            RNA_REF_DB, f_fp,
+                            join(out_dir, '%s.ribosomal.R1' % run_prefix),
+                            join(out_dir, '%s.nonribosomal.R1' % run_prefix),
+                            threads))
+        else:  # there is a forward and reverese seq
+            cmds.append('sortmerna --ref %s --reads %s '
+                        '--aligned %s --other %s '
+                        '--fastx --blast 1 --num_alignments 1 -a %s' % (
+                            RNA_REF_DB, f_fp,
+                            join(out_dir, '%s.ribosomal.R1' % run_prefix),
+                            join(out_dir, '%s.nonribosomal.R1' % run_prefix),
+                            threads))
+
```

Comment:
I think the forward read command can be added always, like:

```suggestion
    for run_prefix, sample, f_fp, r_fp in samples:
        cmds.append('sortmerna --ref %s --reads %s '
                    '--aligned %s --other %s '
                    '--fastx --blast 1 --num_alignments 1 -a %s' % (
                        RNA_REF_DB, f_fp,
                        join(out_dir, '%s.ribosomal.R1' % run_prefix),
                        join(out_dir, '%s.nonribosomal.R1' % run_prefix),
                        threads))
        # there is a forward and reverese seq
        if r_fp is not None:
```

Reply:
Yep, makes sense. Modified accordingly.